frontend.md

```diff
@@ -1,19 +1,46 @@
 ## Applies to [All languages]
 
-# Wrapper structure
-All wrappers should follow a similar structure to maintain simplicity and readability for all devevelopers working on this project *and* all developers using our wrappers.
-
-**Backend** - How the wrapper works behind the scenes. <br>
-**Frontend** - How the end user interacts with our wrapper.
-
-## Frontend
+# Frontend
 A few general rules for frontend:
+--
+
  - Languages **should** follow their most used naming conventions separately.</br></br>
  - All classes/structs/objects used in the front end **should** be prefixed with "Hypixel" e.g. `HypixelPlayer` `HypixelGuild` `HypixelSkyblockProfile` (there may be exceptions, these will be documented per language)
  - Between languages, all classes/structs/objects **should** have similar names. If you are unsure about what names to use, look at the source code of other wrappers.</br></br>
  - Between languages, all methods of corresponding classes/structs/objects **should** have similar names. These methods **must** return the same data.
  - Methods **must** always be provided with documentation describing what the method returns.
 
-### Need help?
+Detailed information for frontend:
+--
+- A main data structure called `HypixelAPI` **must** exist in the wrapper. `HypixelAPI` **must** store the following data:
+
+| name | type | note |
+| --- | --- | --- |
+| API Key | **`UUID`** | This is a key retrieved from Hypixel and provided by the developer. |
+| Cache | **`HypixelCache`** | This may be a reference. `HypixelCache` is an interface, see below. |
+| Cache Strategy | **`HypixelCacheStrategy`** | This may be a reference. `HypixelCacheStrategy` is an interface, see below. |
```

Comment:
Slightly unclear : what's a cache strategy, and what should it contain?

Reply:
it's not finished yet, I'm polling for the layout and style etc